shiny/express/ui/_cm_components.py

```diff
@@ -1067,17 +1075,16 @@ def navset_bar(
     fillable: bool | list[str] = True,
     gap: Optional[CssUnit] = None,
     padding: Optional[CssUnit | list[CssUnit]] = None,
-    position: Literal[
-        "static-top", "fixed-top", "fixed-bottom", "sticky-top"
-    ] = "static-top",
     header: TagChild = None,
     footer: TagChild = None,
-    bg: Optional[str] = None,
-    # TODO: default to 'auto', like we have in R (parse color via webcolors?)
-    inverse: bool = False,
-    underline: bool = True,
-    collapsible: bool = True,
+    navbar_options: Optional[NavbarOptions] = None,
```

Comment:
I feel somewhat on the fence about this, but would you be open to having `types.NavbarOptions` inherit from `TypedDict`? The benefit being that we wouldn't need `ui.navbar_options()`. FWIW, the `animate` argument of `input_slider()` follows that pattern.

Reply:
Totally open to it and it's certainly a route I thought about taking. It's a little more complicated because there are methods on `NavbarOptions`, so we'd need a user-facing `NavbarOption` that inherits from `TypedDict` and a counterpart `NavbarOptionsResolved` (or similar name) class that resolves the user-facing options.